src/router.js

```diff
@@ -167,6 +167,11 @@ of the `key` attribute.)
           forceReplace(router, store, '/');
       }));
     }
+
+    const key = to.meta.delayedData;
+    unwatch.push(store.watch((state) => state.request.data[key], () => {
+      updateDocumentTitle(to);
+    }));
```

Comment:
There's one other case I just thought of, which is that data can be cleared and re-fetched while the user is on a page. It's pretty uncommon, and the only example I can think of that would be relevant here is enabling encryption: if you enable encryption on a project, the `project` data in the store is cleared, and a new request for the project is sent. Clearing the project will trigger this watcher on the store, which will reset the title. That might actually be the behavior we want — mostly I just wanted to put the scenario on your radar.

Reply:
Got it! Updating the title in this case of encryption changing, even if the project name didn't change, seems ok.